ida-ws/src/main/java/upb/ida/dao/DataRepository.java

```diff
@@ -0,0 +1,591 @@
+		}
+		return null;
+	}
+
+
+	/**
+	 *
+	 * @param dataset - Name of the dataset
+	 * @return - Metadata of the dataset as a map.
+	 */
+	public Map<String, Object> getDataSetMD(String dataset) {
+		Map<String, Integer> classCountMap = new HashMap<>();
+		Set<String> distinctColumns = new TreeSet<>();
+		Map<String, ArrayList<String>> columnMap;
+		Map<String, String> columnCommentMap = new HashMap<>();
+		Map<String, String> columnTypeMap = new HashMap<>();
+		Map<String, String> columnLabelMap = new HashMap<>();
+		Map<String, String> classBaseUrlMap = new HashMap<>();
+		String className;
+		int rowCount;
+		QuerySolution resource;
+		int index;
+		String queryString = IDALiteral.PREFIXES +
+				"SELECT ?class (count(?class) as ?count)\n" +
+				"WHERE {\n" +
+				"  \t?s rdf:type ?class;\n" +
+				"    FILTER (?class != owl:NamedIndividual)\n" +
+				"}\n" +
+				"GROUP BY ?class";
+		ResultSet resultSet = getResultFromQuery(dataset + "-data", queryString);
+		if (resultSet == null) {
+			return null;
+		}
+		while (resultSet.hasNext()) {
+			resource = resultSet.next();
+			className = resource.get("class").asNode().getURI();
+			if (className.contains("#")) {
+				index = className.lastIndexOf("#");
+			} else {
+				index = className.lastIndexOf("/");
+			}
+			className = className.substring(index + 1);
+			rowCount = Integer.parseInt(resource.get("count").asNode().getLiteralValue().toString());
+			classCountMap.put(className, rowCount);
+			classBaseUrlMap.put(className, resource.get("class").asNode().getURI());
+		}
+		columnMap = getClassColumnMap(dataset, distinctColumns);
+		Map<String, ArrayList<String>> columnInfoMap = getColumnsInformation(dataset, distinctColumns);
+		ArrayList<String> columnInfo;
+		for(String colName: columnInfoMap.keySet()){
+			columnInfo = columnInfoMap.get(colName);
+			columnCommentMap.put(colName, columnInfo.get(0));
+			columnTypeMap.put(colName, columnInfo.get(1));
+			columnLabelMap.put(colName, columnInfo.get(2));
+		}
+		Map<String, Object> dsInfo = new HashMap<>();
+		dsInfo.put("dsName", dataset);
+		dsInfo.put("dsDesc", "");
+		List<Map<String, Object>> tables = new ArrayList<>();
+		Map<String, Object> table;
+		String classKey;
+		for (String cls : classCountMap.keySet()) {
+			classKey = cls.replaceAll(" ", "");
+			if (columnMap.get(classKey) != null) {
+				table = new HashMap<>();
+				table.put("displayName", cls);
+				table.put("fileName", cls);
+				table.put("colCount", columnMap.get(classKey).size());
+				table.put("rowCount", classCountMap.get(cls));
+				table.put("baseUrl", classBaseUrlMap.get(cls));
+				table.put("fileColMd", createColumnsMetadata(columnMap, classKey, dataset, columnCommentMap, columnLabelMap, columnTypeMap));
+				tables.add(table);
+			}
+		}
+		dsInfo.put("filesMd", tables);
+		if (conn != null) {
+			conn.close();
+		}
+		return dsInfo;
+	}
+
+	/**
+	 * @param className - Name of the table.
+	 * @return - list of rows in the table in JSON format.
+	 */
```

Comment:
Same changes as mentioned in previous method

Reply:
Resolved